src/main/java/de/tum/in/www1/exerciseapp/web/rest/ExerciseResource.java

```diff
@@ -164,38 +160,25 @@ public ExerciseResource(ExerciseRepository exerciseRepository, ExerciseService e
      * @return the ResponseEntity with status 200 (OK) and the list of exercises in body
      */
     @GetMapping(value = "/courses/{courseId}/exercises")
-    @PreAuthorize("hasAnyRole('USER', 'TA', 'ADMIN')")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
     @Timed
     public ResponseEntity<Collection<Exercise>> getExercisesForCourse(@PathVariable Long courseId, @RequestParam(defaultValue = "false") boolean withLtiOutcomeUrlExisting, Principal principal) {
         log.debug("REST request to get Exercises for Course : {}", courseId);
 
         List<Exercise> result;
-
-        // first get user and create representative authorities
         User user = userService.getUserWithGroupsAndAuthorities();
-        Authority adminAuthority = new Authority();
-        adminAuthority.setName("ROLE_ADMIN");
-        Authority taAuthority = new Authority();
-        taAuthority.setName("ROLE_TA");
 
-        // get the course
         Course course = courseService.findOne(courseId);
-
-        // determine user's access level for this course
-        if (user.getAuthorities().contains(adminAuthority)) {
-            // user is admin
-            result = exerciseRepository.findByCourseId(courseId);
-        } else if (user.getAuthorities().contains(taAuthority) && user.getGroups().contains(course.getTeachingAssistantGroupName())) {
-            // user is TA for this course
-            result = exerciseRepository.findByCourseId(courseId);
-        } else if (user.getGroups().contains(course.getStudentGroupName())) {
+        if(!authCheckService.isAuthorizedForCourse(course)) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+        if(user.getGroups().contains(course.getStudentGroupName())) {
             // user is student for this course
             result = withLtiOutcomeUrlExisting ? exerciseRepository.findByCourseIdWhereLtiOutcomeUrlExists(courseId, principal) : exerciseRepository.findByCourseId(courseId);
             // filter out exercises that are not released (or explicitly made visible to students) yet
             result = result.stream().filter(Exercise::getIsVisibleToStudents).collect(Collectors.toList());
         } else {
-            // user has no permissions for this course
-            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+            result = exerciseRepository.findByCourseId(courseId);
         }
```

Comment:
are you 100% sure that the logic stays the same with the new service? This is extremely important, because this call is invoked on the start page for all ArTEMiS users and we spent a lot of time to get it right

Reply:
As far as I can judge.. If you look into the new service, it does all the same checks except user.getAuthorities().contains(taAuthority), which I find excessive because we anyway do user.getGroups().contains(course.getTeachingAssistantGroupName()). The new service also adds the check for the instructor role. I have checked for myself (once being admin and once not), and it resolves courses fine, but I obviously can't check for anyone else.